### PR TITLE
Add documentation how Application Broker handles API Packages

### PR DESCRIPTION
**Pull request by klaudiagrz**
**Description**

We must document how the Application Broker workflow looks like in the Compass mode.

Changes proposed in this pull request:

- Update Application Broker architecture document [Compass mode]
- Update Application Broker diagrams [Compass mode]
- Fix typos in the Application Connector documents [legacy mode]

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/7658
https://github.com/kyma-project/kyma/issues/9033

### CODE REVIEW

File: docs/application-connector-compass-mode/02-04-application-broker.md [hunk 1]
```diff
@@ -8,45 +8,60 @@ The Application Broker (AB) workflow consists of the following steps:
...
+### Generating API credentials
+
+This section describes in detail how the API credentials are generated.
+
+1. User creates a ServiceInstance.
+2. Service Catalog sends a provisioning request to the Application Broker.
+3. Application Broker requests credentials for a ServiceInstance. The request is processed through the Director proxy exposed by Runtime Agent to Director.
+>**NOTE:** Director proxy is secured by [Istio RBAC](https://github.com/kyma-project/kyma/blob/master/resources/compass-runtime-agent/templates/istio-rbac.yaml) and only Application Broker can access it.
+
+4. Application Broker polls credential status with 20 minutes timeout. The polling is completed when the credential status is either `SUCCEEDED` or `FAILED`.
```
Comment (majakurcius) [inline docs/application-connector-compass-mode/02-04-application-broker.md:44]:
```suggestion
4. Application Broker polls credential status with a 20 minute timeout. The polling is completed when the credential status is either `SUCCEEDED` or `FAILED`.
```

File: docs/application-connector-compass-mode/02-04-application-broker.md [hunk 2]
```diff
@@ -8,45 +8,60 @@ The Application Broker (AB) workflow consists of the following steps:
...
+![Credentials flow](./assets/api-credentials-flow.svg)
 
-![API Service Class](./assets/005-AB-API-service-class.svg)
+Application Broker uses these credentials during the binding action. To use another set of credentials, you must create a new ServiceInstance.
 
-## Provisioning and binding for an event ServiceClass
+## Provisioning and binding for an event ServicePlan
 
-This ServiceClass has a **bindable** parameter set to `false` which means that after provisioning a ServiceClass in the Namespace, given events are ready to use for all services. The provisioning workflow for an event ServiceClass consists of the following steps:
+This ServicePlan has a **bindable** parameter set to `false` which means that after provisioning a ServiceInstance in a Namespace, given events are ready to use for all services. The provisioning workflow for an event ServicePlan consists of the following steps:
```
Comment (majakurcius) [inline docs/application-connector-compass-mode/02-04-application-broker.md:52]:
```suggestion
This ServicePlan has a **bindable** parameter set to `false`, which means that after provisioning a ServiceInstance in a Namespace, given events are ready to use for all services. The provisioning workflow for an event ServicePlan consists of the following steps:
```

File: docs/application-connector-compass-mode/06-03-event-activation.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ kubectl get crd eventactivations.applicationconnector.kyma-project.io -o yaml
 
 ## Sample custom resource
 
-This is a sample resource that allows you to consume events sent from the service with the `ac031e8c-9aa4-4cb7-8999-0d358726ffaa` planID in a `production` Namespace.
+This is a sample resource that allows you to consume events sent from the service with the `ac031e8c-9aa4-4cb7-8999-0d358726ffaa` ServicePlan ID in a `production` Namespace.
```
Comment (majakurcius) [inline docs/application-connector-compass-mode/06-03-event-activation.md:14]:
```suggestion
This is a sample resource that allows you to consume events sent from the service with the `ac031e8c-9aa4-4cb7-8999-0d358726ffaa` ServicePlan ID in the `production` Namespace.
```
This is a specific Namespace name, so the definite article should be used.

File: docs/application-connector/06-03-event-activation.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ kubectl get crd eventactivations.applicationconnector.kyma-project.io -o yaml
 
 ## Sample custom resource
 
-This is a sample resource that allows you to consume events sent from the service with the `ac031e8c-9aa4-4cb7-8999-0d358726ffaa` ID in a `production` Namespace.
+This is a sample resource that allows you to consume events sent from the service with the `ac031e8c-9aa4-4cb7-8999-0d358726ffaa` ServiceClass ID in a `production` Namespace.
```
Comment (majakurcius) [inline docs/application-connector/06-03-event-activation.md:14]:
```suggestion
This is a sample resource that allows you to consume events sent from the service with the `ac031e8c-9aa4-4cb7-8999-0d358726ffaa` ServiceClass ID in the `production` Namespace.
```
This is a specific Namespace name, so the definite article should be used.

File: docs/application-connector-compass-mode/02-04-application-broker.md [hunk 5]
```diff
@@ -8,45 +8,60 @@ The Application Broker (AB) workflow consists of the following steps:
...
+2. Provision this ServiceClass with the chosen event ServicePlan by creating a ServiceInstance in the given Namespace.
 3. During the provisioning process, the [EventActivation](/components/application-connector/#custom-resource-event-activation) resource is created together with the ServiceInstance. This resource allows you to create a Trigger from the Kyma Console.
 4. The Application sends an event to the Application Connector.
-5. When the Trigger receives the event, it triggers the Function based on the paramertes defined for a Trigger.
+5. When the Trigger receives the event, it triggers the Function based on the parameters defined for a Trigger.
 
 For more information, read about [event processing and delivery](/components/event-mesh/#details-event-processing-and-delivery).
 
-![Event Service Class](./assets/006-AB-event-service-class.svg)
+![Event ServicePlan](./assets/006-AB-event-service-plan.svg)
```
Comment (majakurcius) [inline docs/application-connector-compass-mode/02-04-application-broker.md:62]: Shouldn't we add arrows in the diagram to steps 3, 4, and 5?

File: docs/application-connector-compass-mode/01-01-application-connector.md [hunk 6]
```diff
@@ -2,6 +2,8 @@
 title: Overview
 ---
 
+>**CAUTION:** Documentation in this directory is under development so some information may not be up-to-date.
+
 The Application Connector (AC) is a custom, in-house built Kyma component that allows you to connect with external solutions. No matter if you want to integrate an on-premise or a cloud system, the integration process does not change, which allows to avoid any configuration or network-related problems.
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/application-connector-compass-mode/01-01-application-connector.md:7]: ... or `in-house built` - that's what Google suggests. Anyway, we should also apply the change to the legacy mode docs after we agree upon the option.

File: docs/application-connector-compass-mode/02-04-application-broker.md [hunk 7]
```diff
@@ -8,45 +8,60 @@ The Application Broker (AB) workflow consists of the following steps:
...
+>**NOTE:** If you want to call the Application directly, skipping the Application Gateway component, use **sanitized({API\_NAME}_{API_ID})_TARGET_URL** with the corresponding **CONFIGURATION** and **CREDENTIALS_TYPE**. For more information, read the document about [proxy configuration](https://kyma-project.io/docs/components/application-connector#details-application-gateway-proxy-configuration).
   * ServiceBindingUsage injects the Secret, together with the label given during the registration process, to the Function or service.
-4. The service or Function calls the API through the Application Connector. The Application Connector verifies the label to check if you have the authorization to access this API.
-5. After verifying the label, the Application Connector allows you to access the Application API.
+4. The service or Function calls the API through the Application Gateway, allowing you to access the Application API.
+5. The Application Gateway enriches the call with the necessary credentials, which it fetches from Secrets created by Service Catalog.
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/application-connector-compass-mode/02-04-application-broker.md:32]: Perfect, thank you for clarifying this ❤️

File: docs/application-connector-compass-mode/02-04-application-broker.md [hunk 8]
```diff
@@ -8,45 +8,60 @@ The Application Broker (AB) workflow consists of the following steps:
...
+
+![API ServicePlan](./assets/005-AB-API-service-plan.svg)
+
+### Generating API credentials
+
+This section describes in detail how the API credentials are generated.
+
+1. User creates a ServiceInstance.
+2. Service Catalog sends a provisioning request to the Application Broker.
+3. Application Broker requests credentials for a ServiceInstance. The request is processed through the Director proxy exposed by Runtime Agent to Director.
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/application-connector-compass-mode/02-04-application-broker.md:42]: Haven't we agreed on not preceding the components' names with the definite article?

File: docs/application-connector-compass-mode/02-04-application-broker.md (hunk 5 above)
Comment (klaudiagrz) [reply docs/application-connector-compass-mode/02-04-application-broker.md:63]: I'd add an arrow to step 5, as Trigger is the subject here. In 3 and 4, all the magic happens in the background - that's why we've decided to use the dashed line.

File: docs/application-connector-compass-mode/01-01-application-connector.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/application-connector-compass-mode/01-01-application-connector.md:7]: In that case, I'm fine with it :) Let's go with `in-house built`. It's prettier, too :)

File: docs/application-connector-compass-mode/02-04-application-broker.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/application-connector-compass-mode/02-04-application-broker.md:32]: No problem 🚀

File: docs/application-connector-compass-mode/02-04-application-broker.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/application-connector-compass-mode/02-04-application-broker.md:42]: We have, but we haven't introduced the changes to the documentation yet. We only have an [issue](https://github.com/kyma-project/community/issues/481) for that. I thought staying consistent for now would make sense, wouldn't it?

File: docs/application-connector-compass-mode/02-04-application-broker.md (hunk 5 above)
Comment (majakurcius) [reply docs/application-connector-compass-mode/02-04-application-broker.md:63]: Fine by me :)

File: docs/application-connector-compass-mode/02-04-application-broker.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/application-connector-compass-mode/02-04-application-broker.md:42]: I'm afraid there are already some inconsistencies in docs regarding that aspect. I'd skip `the` in this case, that would leave us less work in the future.

File: docs/application-connector-compass-mode/02-04-application-broker.md [hunk 9]
```diff
@@ -8,45 +8,60 @@ The Application Broker (AB) workflow consists of the following steps:
...
+>**NOTE:** If you want to call the Application directly, skipping the Application Gateway component, use **sanitized({API\_NAME}_{API_ID})_TARGET_URL** with the corresponding **CONFIGURATION** and **CREDENTIALS_TYPE**. For more information, read the document about [proxy configuration](https://kyma-project.io/docs/components/application-connector#details-application-gateway-proxy-configuration).
   * ServiceBindingUsage injects the Secret, together with the label given during the registration process, to the Function or service.
-4. The service or Function calls the API through the Application Connector. The Application Connector verifies the label to check if you have the authorization to access this API.
-5. After verifying the label, the Application Connector allows you to access the Application API.
+4. The service or Function calls the API through the Application Gateway, allowing you to access the Application API.
+5. The Application Gateway enriches the call with the necessary credentials, which it fetches from Secrets created by Service Catalog.
+
+![API ServicePlan](./assets/005-AB-API-service-plan.svg)
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/application-connector-compass-mode/02-04-application-broker.md:34]: After the discussion with Gophers, we've decided to go with the minimal diagram. Thank you for providing us with the diagrams ❤️

[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/application-connector-compass-mode/02-04-application-broker.md:34]: No problem 🚀

File: docs/application-connector-compass-mode/02-04-application-broker.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/application-connector-compass-mode/02-04-application-broker.md:42]: In that case, fine by me.

**Pull request merged**